docs: improve kargo concepts docs (#2564)
akuity/kargo: docs/docs/15-concepts.md

### What is a Stage?

When you hear the term “environment”, what you envision will depend

> significantly on your perspective. For a developer, the term "environment" will
> have a different meaning while for a Devops engineer, the meaning will be different.

Reviewer: I think this can be more clear, still.

```suggestion
significantly on your perspective. A developer, for example, may think of
an "environment" as a specific _instance_ of an application they work on,
while a DevOps engineer, may think of an "environment" as a particular segment
of the infrastructure they maintain.
```

### What is Freight?

> **Freight** is Kargo's second most important concept. Frieight generally refers
> to a collection of artifacts that gets promoted together to the next stage.
> A single "piece of freight" is a set of references to one or more versioned
> artifacts, which may include one or more:

Reviewer: tbh, I feel this change adds more confusion. We should be careful of unnecessarily using words like "generally," as it implies there are exceptions to what you're about to say. Freight isn't _generally_ a collection of artifacts. Freight _is_ a collection of artifacts. There isn't an exception to that.

The bit about Freight being what moves from stage to stage is already stated just a few lines down from here.

imho, I would revert this change.

```suggestion
**Freight** is Kargo's second most important concept. A single "piece of
freight" is a set of references to one or more versioned artifacts, which may
include one or more:
```

### What is a Warehouse?

> A **warehouse** is a _source_ of freight. It is where a collection of 'Freight'
> gets stored before the 'Freight' is deployed anywhere.
> 
> A warehouse subscribes to one or more:

Reviewer: ```suggestion
A **warehouse** is a _source_ of freight. A warehouse subscribes to one or more:
```

### What is a Warehouse?

A **warehouse** is a _source_ of freight. It is where a collection of 'Freight'
gets stored before the 'Freight' is deployed anywhere.

> A warehouse subscribes to one or more:

Reviewer: Even in the real word, I think warehouses are a somewhat fuzzy concept. Is a warehouse a sort of archive where physical things are stored, possibly long-term? Or is it more of a fulfillment/shipping hub? It can be one or the other, or both.

Kargo warehouses are really about fulfillment. I'm not sure it would be accurate to say they _store_ freight. Freight are instances of a CRD. So, really, _Kubernetes_ stores your Freight. The first few sections of this page stay away from technical details. CRDs are not mentioned until later. But I think the statement that warehouses _store_ freight may set the reader up for confusion when they do read about the CRD later.
